alf/utils/summary_utils.py

```diff
@@ -118,3 +143,58 @@ def histogram_continuous(name,
                 a=tf.stack([left_edges, right_edges, bucket_counts]))
         return tf.summary.write(
             tag=tag, tensor=tensor, step=step, metadata=summary_metadata)
+
+
+@_summary_wrapper
+def add_variables_summaries(grads_and_vars, step):
+    """Add summaries for variables.
+
+    Args:
+      grads_and_vars (list): A list of (gradient, variable) pairs.
+      step (tf.Variable): Variable to use for summaries.
+    """
+
+    for grad, var in grads_and_vars:
+        if grad is not None:
+            if isinstance(var, tf.IndexedSlices):
+                var_values = var.values
+            else:
+                var_values = var
+            var_name = var.name.replace(':', '_')
+            tf.summary.histogram(
+                name='summarize_vars/' + var_name + '_value',
+                data=var_values,
+                step=step)
+            tf.summary.scalar(
+                name='summarize_vars/' + var_name + '_value_norm',
+                data=tf.linalg.global_norm([var_values]),
+                step=step)
+
+
+@_summary_wrapper
+def add_gradients_summaries(grads_and_vars, step):
+    """Add summaries to gradients.
+
+    Args:
+      grads_and_vars (list): A list of gradient to variable pairs (tuples).
+      step (tf.Variable): Variable to use for summaries.
+    """
+    for grad, var in grads_and_vars:
+        if grad is not None:
+            if isinstance(grad, tf.IndexedSlices):
+                grad_values = grad.values
+            else:
+                grad_values = grad
+            var_name = var.name.replace(':', '_')
+            tf.summary.histogram(
+                name='summarize_grads/' + var_name + '_gradient',
+                data=grad_values,
+                step=step)
+            tf.summary.scalar(
+                name='summarize_grads/' + var_name + '_gradient_norm',
+                data=tf.linalg.global_norm([grad_values]),
+                step=step)
+
+
+# tf.summary.scalar = _summary_wrapper(tf.summary.scalar)
+tf.summary.histogram = _summary_wrapper(tf.summary.histogram)
```

Comment:
should we do the same for add_gradients_summaries and add_variables_summaries instead of copying them?

Reply:
`add_gradients_summaries` in tfa using `tf.compat.v2.summary` 
so it will be `tf.compat.v2.summary.histogram=_summary_wrapper(tf.compat.v2.summary.histogram)`
we should avoid include `tf.compat` in alf